### PR TITLE
Up variable encoding

### PR DESCRIPTION
**Pull request by HamdaHmida**

### CODE REVIEW

File: skdecide/hub/domain/up/up.py [hunk 1]
```diff
@@ -573,6 +683,36 @@ def _get_observation_space_(self) -> D.T_agent[Space[D.T_observation]]:
                         else np.int32
                     ),
                 )
+            elif self._state_encoding == "variable":
+                self._observation_space = VariableSpace(
+                    Box(
```
Comment (fteicht) [inline skdecide/hub/domain/up/up.py:688]: @HamdaHmida Can you please use the scikit-decide version of gym.Box, i.e. BoxSpace?

File: skdecide/hub/domain/up/up.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 import numpy as np
 import unified_planning as up
+from gymnasium.spaces import Box
```
Comment (fteicht) [inline skdecide/hub/domain/up/up.py:9]: @HamdaHmida Please use only the scikit-decide version, i.e. BoxSpace, which is already imported a few lines below.

Comment (HamdaHmida) [reply skdecide/hub/domain/up/up.py]: @fteicht j'ai fait le changement demandé et j'ai met à jour la branche

**Review by fteicht — APPROVED** [review]
Thanks a lot @HamdaHmida , great job!

**Comment by HamdaHmida** [comment]
@fteicht Welcome.

**Pull request merged**